dm/dm/master/openapi_controller.go

```diff
@@ -357,60 +355,64 @@ func (s *Server) checkTask(ctx context.Context, subtaskCfgList []*config.SubTask
 	return checker.CheckSyncConfigFunc(ctx, subtaskCfgList, errCnt, warnCnt)
 }
 
-func (s *Server) checkOpenAPITaskBeforeOperate(ctx context.Context, task *openapi.Task) ([]*config.SubTaskConfig, error) {
+func (s *Server) checkOpenAPITaskBeforeOperate(ctx context.Context, task *openapi.Task) ([]*config.SubTaskConfig, string, error) {
 	// prepare target db config
 	toDBCfg := config.GetTargetDBCfgFromOpenAPITask(task)
 	if err := adjustTargetDB(ctx, toDBCfg); err != nil {
-		return nil, err
+		return nil, "", err
 	}
 	// prepare source db config source name -> source config
 	sourceCfgMap := make(map[string]*config.SourceConfig)
 	for _, cfg := range task.SourceConfig.SourceConf {
 		if sourceCfg := s.scheduler.GetSourceCfgByID(cfg.SourceName); sourceCfg != nil {
 			sourceCfgMap[cfg.SourceName] = sourceCfg
 		} else {
-			return nil, terror.ErrSchedulerSourceCfgNotExist.Generate(sourceCfg.SourceID)
+			return nil, "", terror.ErrSchedulerSourceCfgNotExist.Generate(sourceCfg.SourceID)
 		}
 	}
 	// generate sub task configs
 	subTaskConfigList, err := config.OpenAPITaskToSubTaskConfigs(task, toDBCfg, sourceCfgMap)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 	// check subtask config
 	msg, err := s.checkTask(ctx, subTaskConfigList, common.DefaultErrorCnt, common.DefaultWarnCnt)
 	if err != nil {
-		return nil, terror.WithClass(err, terror.ClassDMMaster)
-	}
-	if len(msg) != 0 {
-		// TODO: return warning msg with http.StatusCreated and task together
-		log.L().Warn("openapi pre-check warning before start task", zap.String("warning", msg))
+		return nil, "", terror.WithClass(err, terror.ClassDMMaster)
 	}
-	return subTaskConfigList, nil
+	return subTaskConfigList, msg, nil
 }
 
-func (s *Server) createTask(ctx context.Context, req openapi.CreateTaskRequest) (*openapi.Task, error) {
+func (s *Server) createTask(ctx context.Context, req openapi.CreateTaskRequest) (*openapi.OperateTaskResponse, error) {
 	task := &req.Task
 	if err := task.Adjust(); err != nil {
 		return nil, err
 	}
-	subTaskConfigList, err := s.checkOpenAPITaskBeforeOperate(ctx, task)
+	subTaskConfigList, msg, err := s.checkOpenAPITaskBeforeOperate(ctx, task)
```

Comment:
Are there any logical errors included in `err` (e.g. some typo from users' config), which causes the failure of `CheckTask`? If so, the `err` should be returned to the front-end interface. Or, it merely contains internal errors (e.g. worker's offline), then it's okay to not return.

Reply:
if there has error, it will be returned. and `msg` is the message about waring info or success info. error will be handled in [openapi_view](https://github.com/pingcap/tiflow/pull/5240/files#diff-7d9499db50ffd3895e6f45ab0aee935dac04929a8e4a200d584c37331746d095L434).